### ISSUE TITLE
auto_discover_path should support subfolders

### ISSUE THREAD

**Issue by uwe42** [intelsdi-x/snap#1476]
If auto_discover supports plugins and tasks, i would like to keep them separated. Then i can write scripts to pull new plugins and remove or edit the complete folder without touching the task file(s) (or move them around).

E.g. after a download of the plugins i would like to do something like chmod +x autoDiscover/plugins/*

**Comment by candysmurf** [comment]
@uwe42, plugin binary type is different than task files. You can do what you wanted today. May I know any other reasons that you like tasks are in a different folder? If putting tasks in the subfolder is acceptable?

**Comment by uwe42** [comment]
@candysmurf, it would be nice to have a more ordered structure of the plugins and tasks loaded on start. It would be "nice to have" to keep things that do not belong to each other separated in a structure. Other possibility would be to extend the config file for a separated auto_discover_plugin / auto_discover_task, so no sub-folders and i can separate them through the config file.

**Comment by candysmurf** [comment]
@uwe42, agreed, thanks for your suggestions. I'll label it as an enhancement.

**Comment by mjbrender** [comment]
Love this proposal @uwe42. Very cool that you proposed it. We'll keep you posted! 🐢
